### PR TITLE
Add redeem instructions

### LINKED ISSUE #268: Instruction for redemption pages

### PR DESCRIPTION
**Pull request by zognin**
Closes #268

Can someone help test whether the ? button works in mobile because sometimes it works and sometimes it doesn't for me
Can someone help check the aspect ratio on big screens too? Cos I don't have a big screen.

Notes
- Haven't added animations of instructions opening and closing
- Slide content can prob change later, for now it's just the bare min
- The ? button for mobile should be in the nav bar but for now it's like that because it's going to take a bit of work to do that so it's prob better to do that in another PR

### CODE REVIEW

File: frontend/components/redeem/Receipt.tsx [hunk 1]
```diff
@@ -48,13 +48,13 @@ interface ReceiptItemProps {
...
     <Grid container alignItems="center">
-      <Grid item xs={1}>
+      <Grid item xs={2}>
         <Avatar variant="rounded">
           {imageBaseUrl ? <Box sx={imageSx} component="img" src={imageBaseUrl} /> : <AccountBoxIcon />}
         </Avatar>
       </Grid>
 
-      <Grid item xs={11} padding={2}>
+      <Grid item xs={10} padding={2}>
```
Comment (sivayogasubramanian) [inline frontend/components/redeem/Receipt.tsx:57]:
```suggestion
      <Grid item xs={10} padding={2} marginLeft={1}>
```

File: frontend/pages/redeem/[urlToken]/index.tsx [hunk 2]
```diff
@@ -136,6 +143,15 @@ const Redeem: NextPage = () => {
...
 
       <Container component="main" maxWidth="md">
+        {isCouponNotRedeemed && (
+          <IconButtonWithTooltip
+            sx={isMobile ? mobileHelpButtonSx : desktopHelpButtonSx}
+            icon={<HelpOutlineIcon />}
+            tooltip="Instructions"
+            onClick={() => setOpenInstructions(true)}
+          />
+        )}
```
Comment (sivayogasubramanian) [inline frontend/pages/redeem/[urlToken]/index.tsx:153]: The button does not work for me for mobile view.


https://user-images.githubusercontent.com/53614454/197392299-66637369-7f13-4021-9d2f-320a9e4079d7.mp4

Comment (sivayogasubramanian) [reply frontend/pages/redeem/[urlToken]/index.tsx:153]: image

The reason is the stepper is covering the button and the stepper has zIndex: 1.

Easy fix by setting zIndex of button to be > 1

File: frontend/components/redeem/instructions/InstructionsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  dialogPaperSx,
+  mobileScreenDisplaySx,
+  mobileSlideSx,
+  mobileSwiperSx,
+  slideContainerSx,
+} from '../../../styles/components/redeem/InstructionDialogStyles';
+import { Box, Stack, useTheme } from '@mui/system';
+import { Navigation, Pagination } from 'swiper';
+import { Swiper, SwiperSlide } from 'swiper/react';
+
```
Comment (sivayogasubramanian) [inline frontend/components/redeem/instructions/InstructionsDialog.tsx:17]: Remove line break

**Review by sivayogasubramanian — APPROVED** [review]
LGTM!

This is what it looks like on a big screen

https://user-images.githubusercontent.com/53614454/197392708-f53690e9-2295-4466-9910-47be3ac31de3.mp4

I think the instructions can just be a normal modal in desktop?

File: frontend/components/redeem/Receipt.tsx (hunk 1 above)
Comment (zognin) [reply frontend/components/redeem/Receipt.tsx:57]: Adopting a different fix as explained.

File: frontend/components/redeem/Receipt.tsx [hunk 4]
```diff
@@ -48,13 +48,13 @@ interface ReceiptItemProps {
 const ReceiptItem = ({ imageBaseUrl, text }: ReceiptItemProps) => {
   return (
     <Grid container alignItems="center">
-      <Grid item xs={1}>
+      <Grid item xs={2}>
```
[the remark this answers is not in the document]
Comment (zognin) [reply frontend/components/redeem/Receipt.tsx:51]: Same as above

**Pull request merged**